src/main/java/book/store/onlinebookstore/model/ShoppingCart.java

```diff
@@ -0,0 +1,36 @@
+package book.store.onlinebookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
+@Entity
+@SQLDelete(sql = "UPDATE sopping_carts set is_deleted = TRUE WHERE id = ?")
+@Where(clause = "is_deleted = FALSE")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne
+    @JoinColumn(name = "user_id", referencedColumnName = "id")
+    private User user;
+
+    @OneToMany(mappedBy = "shoppingCart")
+    private Set<CartItem> cartItems;
+
+    @Column(nullable = false)

```

Comment:
```suggestion
    @Column(nullable = false, name= name = "is_deleted")

```
cause hibernate want to match this value with isDeleted column